pkgs/development/node-packages/aliases.nix

```diff
@@ -0,0 +1,38 @@
+lib: self: super:
+
+### Deprecated aliases - for backward compatibility
+
+with self;
+
+let
+  # Removing recurseForDerivation prevents derivations of aliased attribute
+  # set to appear while listing all the packages available.
+  removeRecurseForDerivations = alias: with lib;
+    if alias.recurseForDerivations or false
+    then removeAttrs alias ["recurseForDerivations"]
+    else alias;
+
+  # Disabling distribution prevents top-level aliases for non-recursed package
+  # sets from building on Hydra.
+  removeDistribute = alias: with lib;
+    if isDerivation alias then
+      dontDistribute alias
+    else alias;
+
+  # Make sure that we are not shadowing something from node-packages.nix.
+  checkInPkgs = n: alias:
+    if builtins.hasAttr n super
+    then throw "Alias ${n} is still in node-packages.nix"
+    else alias;
+
+  mapAliases = aliases:
+    lib.mapAttrs (n: alias:
+      removeDistribute
+        (removeRecurseForDerivations
+          (checkInPkgs n alias)))
+      aliases;
+in
+
+mapAliases ({
+  "@githubnext/github-copilot-cli" = throw "'nodePackages.@githubnext/github-copilot-cli' has been promoted to a top-level 'github-copilot-cli' attribute"; # Added 2023-05-02
```

Comment:
Can we add none throw aliases here, too?

Reply:
What do you mean?

Reply:
I think @SuperSandro2000 means non-throw aliases. Like python-aliases, you can only alias to other names within the set

I can add a `pkgs` argument to the file, though, since this is somewhat a special case for aliasing, and allow aliasing outside of the set if we really want

I'm mostly indifferent so how do you two feel about that?

Reply:
> I can add a `pkgs` argument to the file, though, since this is somewhat a special case for aliasing, and allow aliasing outside of the set if we really want

Yes, that would be really nice. Normally aliases just continue to work for one release and are then converted into a throw.

Reply:
I've pushed an update to allow aliasing outside of the package set now!

> Normally aliases just continue to work for one release and are then converted into a throw.

(Side note, but I left it as I did initially since python-aliases.nix seemed to just immediately make it a throw if the package moved outside of the set. This is more of a special case though since ideally everything just gets moved out of the `nodePackages` set, so having non-throw, non-local aliases makes sense)